src/components/TabGroup/Tab.tsx

```diff
@@ -44,25 +53,38 @@ export const StyledTab = styled.span`
   }
 `
 
-const Tab = ({
+export type TabProps = {
+  as?: ElementType
+  children?: ReactNode
+  disabled?: boolean
+  hasEndedCount?: boolean
+  index?: number
+  isSelected?: boolean
+  isTabsWidthSet?: boolean
+  name?: string
+  onClick?: (
+    event: MouseEvent<HTMLSpanElement> | KeyboardEvent<HTMLSpanElement>,
+  ) => void
+  setInternTabsWidth?: (width: number, index: number) => void
+}
+
+const Tab: FunctionComponent<TabProps> = ({
   children,
-  disabled,
-  isSelected,
+  disabled = false,
+  isSelected = false,
   setInternTabsWidth,
-  isTabsWidthSet,
-  index,
+  isTabsWidthSet = false,
+  index = 0,
   name,
   onClick,
-  hasEndedCount,
+  hasEndedCount = false,
   as,
   ...props
 }) => {
-  const ref = useRef({})
+  const ref = useRef<HTMLSpanElement>(null)
```

Comment:
if we change as="div", the ref could be HTMLDivElement for exemple ?

Reply:
I'm not sure it is usefull to make it more generic...

```suggestion
  const ref = useRef<HTMLElement>(null)
```
And way too complex to have dynamic typing here.